autoPyTorch/data/tabular_feature_validator.py

```diff
@@ -189,10 +204,6 @@ def transform(
         if hasattr(X, "iloc") and not scipy.sparse.issparse(X):
             X = cast(pd.DataFrame, X)
 
-            # Also remove the object dtype for new data
-            if not X.select_dtypes(include='object').empty:
-                X = self.infer_objects(X)
```

Comment:
why do we not want to infer objects here? I think its needed to check if the datatype for a column has changed between train and test. See [this](https://github.com/automl/Auto-PyTorch/blob/217c38dd43b012d3839f757cce7877299aac74cc/autoPyTorch/data/tabular_feature_validator.py#L458)

Reply:
To my understanding it is done a few lines of code later in `_check_data`, here: 

https://github.com/automl/Auto-PyTorch/blob/217c38dd43b012d3839f757cce7877299aac74cc/autoPyTorch/data/tabular_feature_validator.py#L284

Reply:
okay so there were duplications.

Reply:
@ravinkohli yes there were duplicates. @nabenabe0928 the ` infer_objects` is already called in a later function named _check_data, so it did not make sense to have it called before too.

Reply:
Answer below.

Reply:
Ah ok
So this function comes first then